resources/build/trigger-definitions.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+#
+# This file maps specific paths to build triggers
+#
+
+# the base folder for each pattern does not need to be included, nor oem folders
+# e.g. android='common/predictive-text|common/lexical-model-types'
+# will expand into android='^(android|(oem/[^/]+/android)|common/predictive-text|common/lexical-model-types)'
+
+watch_android='web|common/predictive-text|common/lexical-model-types'
+watch_ios='web|common/predictive-text|common/lexical-model-types'
+watch_linux='common/engine'
+watch_mac='common/engine'
+watch_web='common/predictive-text|common/lexical-model-types'
```

Comment:
Wouldn't it be best to define `watch_android` and `watch_ios` more like this?

```
watch_android='web|$watch_web'
watch_ios='web|$watch_web'
```

Or is there some reason we can't do this?

Could also go further with:

```
watch_web='common/predictive-text|common/lexical-model-types'

watch_web_core='web|$watch_web'
watch_android='$watch_web_core'
watch_ios='$watch_web_core'

watch_keyman_core='common/engine'
watch_linux='$watch_keyman_core'
watch_mac='$watch_keyman_core'

# While watch_web_core and watch_keyman_core currently are all that exist in common,
# keeping it explicit below should help future-proof things.
watch_windows='$watch_web_core|$watch_keyman_core|common|developer'
```

Reply:
We could do that but it actually gets a bit uglier to read, because we have to escape the vertical bars...  e.g.

```
watch_ios="web\|$watch_web"
```

And while it shortens the definitions, they are all in one place anyway so it actually makes it slightly *less* easy to see what is triggered where, so for now I'd prefer to keep it a bit WETter. These are optimisations that are easy enough to apply in the future if it becomes unwieldy.